packages/core/src/dialects/mssql/query-generator-typescript.ts

```diff
@@ -50,6 +55,72 @@ export class MsSqlQueryGeneratorTypeScript extends AbstractQueryGenerator {
     ]);
   }
 
+  addConstraintQuery(tableName: TableNameOrModel, options: AddConstraintQueryOptions) {
+    return joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(tableName),
+      'ADD',
+      this.getConstraintSnippet(tableName, options),
+    ]);
+  }
+
+  removeConstraintQuery(tableName: TableNameOrModel, constraintName: string, options?: RemoveConstraintQueryOptions) {
+    if (options) {
+      rejectInvalidOptions(
+        'removeConstraintQuery',
+        this.dialect.name,
+        REMOVE_CONSTRAINT_QUERY_SUPPORTABLE_OPTIONS,
+        REMOVE_CONSTRAINT_QUERY_SUPPORTED_OPTIONS,
+        options,
+      );
+    }
+
+    return joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(tableName),
+      'DROP CONSTRAINT',
+      options?.ifExists ? 'IF EXISTS' : '',
+      this.quoteIdentifier(constraintName),
+    ]);
+  }
+
+  showConstraintsQuery(tableName: TableNameOrModel, constraintName?: string) {
+    const table = this.extractTableDetails(tableName);
+
+    return joinSQLFragments([
+      'SELECT DB_NAME() AS constraintCatalog,',
+      's.[name] AS constraintSchema,',
+      'c.constraintName,',
+      `REPLACE(LEFT(c.constraintType, CHARINDEX('_CONSTRAINT', c.constraintType) - 1), '_', ' ') AS constraintType,`,
+      'DB_NAME() AS tableCatalog,',
+      's.[name] AS tableSchema,',
+      't.[name] AS tableName,',
+      'c.columnNames,',
+      'c.referencedTableName,',
+      'c.referencedColumnNames,',
+      'c.deleteAction,',
+      'c.updateAction,',
+      'c.definition',
+      'FROM sys.tables t INNER JOIN sys.schemas s ON t.schema_id = s.schema_id',
+      'INNER JOIN (',
+      'SELECT [name] AS constraintName, [type_desc] AS constraintType, [parent_object_id] AS constraintTableId, null AS columnNames, null AS referencedTableName',
+      ', null AS referencedColumnNames, null AS deleteAction, null AS updateAction, null AS definition FROM sys.key_constraints UNION ALL',
```

Comment:
why are you selecting null? :o

Reply:
For UNION ALL to work, [you need to have the same number and the order of the columns in all queries](https://learn.microsoft.com/en-us/sql/t-sql/language-elements/set-operators-union-transact-sql?view=sql-server-ver16).
Therefore I decided on using null where the column doesn't exist.